src/transformers/models/vqgan/modeling_vqgan.py

```diff
@@ -0,0 +1,763 @@
+        return hidden_states
+
+
+class ResnetBlock(nn.Module):
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int = None,
+        use_conv_shortcut: bool = False,
+        dropout_prob: float = 0.0,
+    ):
+        super().__init__()
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.out_channels_ = self.in_channels if self.out_channels is None else self.out_channels
+        self.use_conv_shortcut = use_conv_shortcut
+
+        self.activation = SiLUActivation()
+
+        self.norm1 = nn.GroupNorm(num_groups=32, num_channels=in_channels, eps=1e-6, affine=True)
+        self.conv1 = nn.Conv2d(
+            self.in_channels,
+            self.out_channels_,
+            kernel_size=3,
+            stride=1,
+            padding=1,
+        )
+
+        self.norm2 = nn.GroupNorm(num_groups=32, num_channels=self.out_channels_, eps=1e-6, affine=True)
+        self.dropout = nn.Dropout(dropout_prob)
+        self.conv2 = nn.Conv2d(
+            self.out_channels_,
+            self.out_channels_,
+            kernel_size=3,
+            stride=(1, 1),
+            padding=1,
+        )
+
+        if self.in_channels != self.out_channels_:
+            if use_conv_shortcut:
+                self.conv_shortcut = nn.Conv2d(
+                    self.in_channels,
+                    self.out_channels_,
+                    kernel_size=3,
+                    stride=1,
+                    padding=1,
+                )
+            else:
+                self.nin_shortcut = nn.Conv2d(
+                    self.in_channels,
+                    self.out_channels_,
+                    kernel_size=1,
+                    stride=1,
+                    padding=0,
+                )
+
+    def forward(self, hidden_states):
+        residual = hidden_states
+        hidden_states = self.norm1(hidden_states)
+        hidden_states = self.activation(hidden_states)
+        hidden_states = self.conv1(hidden_states)
+
+        hidden_states = self.norm2(hidden_states)
+        hidden_states = self.activation(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+        hidden_states = self.conv2(hidden_states)
+
+        if self.in_channels != self.out_channels_:
+            if self.use_conv_shortcut:
+                residual = self.conv_shortcut(residual)
+            else:
+                residual = self.nin_shortcut(residual)
+
+        return hidden_states + residual
+
+
+class AttnBlock(nn.Module):
+    def __init__(self, in_channels: int):
+        super().__init__()
+
+        self.in_channels = in_channels
+        conv = partial(nn.Conv2d, self.in_channels, self.in_channels, kernel_size=1, stride=1, padding=0)
+
+        self.norm = nn.GroupNorm(num_groups=32, num_channels=self.in_channels, eps=1e-6, affine=True)
+        self.q, self.k, self.v = conv(), conv(), conv()
```

Comment:
No convolution layers anymore for attention blocks please. We have a working solution in `diffusers` that makes use of `nn.Linear` -> let's use this instead